restapi/src/main/java/io/stargate/web/resources/Converters.java

```diff
@@ -1031,4 +1036,129 @@ public static String maybeQuote(String text) {
       return text;
     return '"' + PATTERN_DOUBLE_QUOTE.matcher(text).replaceAll(ESCAPED_DOUBLE_QUOTE) + '"';
   }
+
+  public static Map<String, String> convertColumnMap(UserDefinedType udt) {
+    Map<String, String> map = new HashMap<>();
+    for (Map.Entry<String, Column> e : udt.columnMap().entrySet()) {
+      map.put(e.getKey(), e.getValue().toString());
+    }
+    return map;
+  }
+
+  public static List<Column> fromUdtAdd(UdtAdd udtAdd) {
+    List<UdtType> fieldsList = udtAdd.getFields();
+    if (fieldsList == null || fieldsList.isEmpty()) {
+      throw new IllegalArgumentException("Must have at least one field");
+    }
+    List<Column> fields = new ArrayList<>(fieldsList.size());
+
+    for (UdtType field : fieldsList) {
+      if (Strings.isNullOrEmpty(field.getName())) {
+        throw new IllegalArgumentException(
+            "UDT definition should contain a 'name' field specifying the UDT field name.");
+      }
+      fields.add(Column.create(field.getName(), decodeType(field)));
+    }
+    return fields;
+  }
+
+  public static Column.ColumnType decodeType(UdtType udtType) {
+    CQLType basic = udtType.getBasic();
+    UdtInfo info = udtType.getInfo();
+    List<UdtType> subTypes = info != null ? info.getSubTypes() : null;
+    boolean frozen = info != null ? info.isFrozen() : false;
+
+    switch (basic) {
+      case INT:
+        return Column.Type.Int;
+      case INET:
+        return Column.Type.Inet;
+      case TIMEUUID:
+        return Column.Type.Timeuuid;
+      case TIMESTAMP:
+        return Column.Type.Timestamp;
+      case BIGINT:
+        return Column.Type.Bigint;
+      case TIME:
+        return Column.Type.Time;
+      case DURATION:
+        return Column.Type.Duration;
+      case VARINT:
+        return Column.Type.Varint;
+      case UUID:
+        return Column.Type.Uuid;
+      case BOOLEAN:
+        return Column.Type.Boolean;
+      case TINYINT:
+        return Column.Type.Tinyint;
+      case SMALLINT:
+        return Column.Type.Smallint;
+      case ASCII:
+        return Column.Type.Ascii;
+      case DECIMAL:
+        return Column.Type.Decimal;
+      case BLOB:
+        return Column.Type.Blob;
+      case VARCHAR:
+      case TEXT:
+        return Column.Type.Text;
+      case DOUBLE:
+        return Column.Type.Double;
+      case COUNTER:
+        return Column.Type.Counter;
+      case DATE:
+        return Column.Type.Date;
+      case FLOAT:
+        return Column.Type.Float;
+      case LIST:
+        if (info == null) {
+          throw new IllegalArgumentException(
+              "List cqlType should contain an 'info' field specifying the sub cqlType");
+        }
+        if (subTypes == null || subTypes.size() != 1) {
+          throw new IllegalArgumentException("List sub types should contain 1 item");
+        }
+        return Column.Type.List.of(decodeType(subTypes.get(0))).frozen(frozen);
```

Comment:
Technically, in `List.of(Int)` for example, `Int` is not a sub-type. JRE calls them "Type Parameter".

I personally find the term "sub-type" confusing in this context. I do not insist on "type parameter" but could related code be refactored to avoid "sub-type" in this context? thx!

Reply:
Please, see if the changes I've done suffice. 🙂